### ISSUE TITLE
Provide Option to Truncate Attribute Values that Exceed NR Length Limit

### ISSUE THREAD

**Issue by chase-miller** [newrelic/newrelic-logenricher-dotnet#109]
## Is your feature request related to a problem? Please describe.
New Relic's logging capability is unreliable from a user's perspective when messages aren't parsed (https://docs.newrelic.com/docs/logs/log-management/troubleshooting/log-message-truncated/). I would much rather have truncated, structured data than no structured data at all. Otherwise searches result in false negatives (results are erroneously omitted).

## Feature Description
Provide an option when calling `.Enrich.WithNewRelicLogsInContext()` to truncate any attribute values when length limits would violate New Relic's field length limit of 4096 characters.

## Describe Alternatives
I'm guessing that this would probably be easier if New Relic had a sink instead of an enricher, but hopefully not. Perhaps this would be better placed elsewhere in Serilog's configuration, I'd argue it's best placed in New Relic's Serilog integration due to it being a New Relic-specific issue (albeit also prevalent across other logging providers). 

## Additional context
N/A

## Priority
Please help us better understand this feature request by choosing a priority from the following options: 
[Really Want]

**Comment by nr-ahemsath** [comment]
Hi @chase-miller, thanks for bringing this to our attention.  At first glance, this definitely seems like a solvable problem.

I am confused, however: your problem description (and feature description) imply that when you log a message in Serilog with the New Relic enricher and formatter, and then forward the resulting JSON data to New Relic using one of our supported forwarding mechanisms (e.g. `fluentd`), that any attributes greater than 4096 chars in length will not show up in the NR logs UI.  However, the NR doc you linked (https://docs.newrelic.com/docs/logs/log-management/troubleshooting/log-message-truncated/) says that "Any data longer than [4096 chars] is truncated during ingestion." and implies that the partial data <=4096 chars in length will be shown with an ellipsis (...) at the end.  So it seems like if our doc is correct, the backend already does what you're asking our library in this repo to do.  If our doc is incorrect, we should address that as well as implement the feature you're asking for.

(Or maybe I'm not understanding the problem, in which case I apologize in advance!)

If we do need to modify the behavior of our library, I think we can most easily do it in our formatter class instead of the enricher: https://github.com/newrelic/newrelic-logenricher-dotnet/blob/a016be5088944a0c33a4cd3a47ebdfa9ee21d6a2/src/NewRelic.LogEnrichers.Serilog/NewRelicFormatter.cs#L186
And my first instinct is that truncating fields to the 4096 maximum length should be the default behavior rather than something required to be configured, since the whole point of the New Relic formatter for Serilog is to produce log output in the JSON format needed by the NR backend.

**Comment by chase-miller** [comment]
@nr-ahemsath thanks for getting back so quickly! 

I perused the code after opening this issue, and I agree that the formatter makes the most sense. I'm currently working on a wrapper around `NewRelicFormatter` to do this before it hits your code, but it would be way more convenient if it were already done ;-). Also I'm not sure if it's even possible. My fallback is to copy `NewRelicForamtter.cs` and modify it. Modifying the method you pointed out makes the most sense to me.

> partial data <=4096 chars in length will be shown with an ellipsis (...)
This is the behavior that's undesirable to me. I could be wrong, but it appears that if one property within a log message exceeds 4096 characters, then the *entire* log message doesn't get parsed. I'd much rather truncate the offending property (e.g. with ellipsis) and have the log message parsed.

Does that makes sense? I might not understand how this works by the way ;-)

**Comment by chase-miller** [comment]
Also, check out [this convo](https://stackoverflow.com/a/39460300/1072030) and the code he links to for an example of how the seq sink is doing something similar.

**Comment by nr-ahemsath** [comment]
We should probably manually test the current behavior to see what it is before doing a fix (and we'd need to manually test after the fix anyway).

I'll move this issue into our backlog.  Since we have a small team and we maintain multiple products, it might be a while before we can get this issue added to a milestone.  If you end up modifying your own `NewRelicFormatter` and that works for you, we'd welcome a PR with your changes.
